portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +24,40 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+
+  private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Spongebob");
+    messages.add("Gary");
+    messages.add("Mr. Krabs");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String message = messages.get((int) (Math.random() * messages.size()));
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(message);
+
+    // Convert the server stats to JSON
+    Output output = new Output();
+    String json = convertToJson(output);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(Output output) {
+    Gson gson = new Gson();
+    String json = gson.toJson(output);
+    return json;
```

Comment:
You can inline all of this above I believe

String json = new Gson().toJson(output)

And then you dont need this method at all.